core/controllers/improvements.py

```diff
@@ -50,23 +79,13 @@ def get(self, exploration_id):
 
     @acl_decorators.can_edit_exploration
     def post(self, exploration_id):
-        task_entries = self.payload.get('task_entries', None)
-        if task_entries is None:
-            raise self.InvalidInputException('No task_entries provided')
+        task_entry_dicts = self.normalized_payload.get('task_entries')
```

Comment:
Same issue -- the key should match what is retrieved. "task_entries" sounds like the domain objects.

Reply:
Updated.